web/client/actions/security.js

```diff
@@ -67,10 +68,15 @@ function logoutWithReload() {
 }
 
 function login(username, password) {
-    return (dispatch) => {
-        AuthenticationAPI.login(username, password).then((response) => {
+    return (dispatch, state) => {
+        return AuthenticationAPI.login(username, password).then((response) => {
```

Comment:
This refreshes can go easily in an epic. Doing it here can cause unwanted side effects.

You should monitor the login/logout actions to reload MapInfo, until you change page.